tensorflow/compiler/xla/service/gpu/ir_emitter.cc

```diff
@@ -205,8 +205,9 @@ bool IrEmitter::MaybeEmitDirectAtomicOperation(
   }
 
   if (root_opcode == HloOpcode::kAdd) {
+    llvm::Triple target_triple = llvm::Triple(module_->getTargetTriple());
     // NVPTX supports atomicAdd on F32 and integer types.
-    if (element_type == F32) {
+    if (target_triple.isNVPTX() && (element_type == F32)) {
```

Comment:
Do AMDGPU support atomic intrinsics? Does it make sense to add a TODO to support them here?

Reply:
There are atomic intrinsics but they don't support floating point types. So we really can't put a TODO here.